projects/igniteui-angular/src/lib/progressbar/circularbar.component.spec.ts

```diff
@@ -1,361 +1,144 @@
+    let fixture: ComponentFixture<IgxCircularProgressBarComponent>;
+    let progress: IgxCircularProgressBarComponent;
+    let circularBar: HTMLElement;
 
-describe('IgxCircularBar', () => {
     configureTestSuite();
 
-    describe('Unit Tests', () => {
-        let progress: IgxCircularProgressBarComponent;
-        let fixture: ComponentFixture<IgxCircularProgressBarComponent>;
-
-        beforeEach(() => {
-            fixture = TestBed.createComponent(IgxCircularProgressBarComponent);
-            progress = fixture.componentInstance;
-            progress.animate = false;
-            fixture.detectChanges();
-        });
-
-        it('Initialize circularProgressbar with default values', () => {
-            const defaultMaxValue = 100;
+    beforeAll(waitForAsync(() => {
+        TestBed.configureTestingModule({
+            imports: [IgxCircularProgressBarComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [IgxCircularProgressBarComponent]
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(IgxCircularProgressBarComponent);
+        progress = fixture.componentInstance;
+        fixture.detectChanges();
+        circularBar = fixture.debugElement.nativeElement;
+    });
 
-            expect(progress.id).toContain('igx-circular-bar-');
-            expect(progress.max).toBe(defaultMaxValue);
-            expect(progress.value).toBe(0);
-        });
+    function clasListContains(element: HTMLElement, className: string, expected: boolean) {
+        expect(element.classList.contains(className)).toBe(expected);
+    }
 
-        it('should set value to 0 for negative numbers', () => {
-            const negativeValue = -20;
-            const expectedValue = 10;
-            expect(progress.value).toBe(0);
+    it('should initialize with default attributes', () => {
+        expect(progress.cssClass).toBe('igx-circular-bar');
+        expect(progress.textVisibility).toBe(true);
+        expect(progress.hasText).toBe(false);
+    });
 
-            progress.value = expectedValue;
-            expect(progress.value).toBe(expectedValue);
+    it('should correctly apply the ID attribute', () => {
+        expect(progress.id).toContain('igx-circular-bar-'); // Keep this
 
-            progress.value = negativeValue;
-            expect(progress.value).toBe(expectedValue);
-        });
+        const customId = 'custom-circular-bar-id';
+        progress.id = customId;
+        fixture.detectChanges();
 
-        it('should set value to 0 for negative numbers', () => {
-            const negativeValue = -20;
-            const expectedValue = 0;
+        expect(progress.id).toBe(customId);
+    });
 
-            progress.value = negativeValue;
+    it('should correctly toggle the indeterminate mode', () => {
+        clasListContains(circularBar, 'igx-circular-bar--indeterminate', false);
```

Comment:
```suggestion
        clasListContains(circularBar, 'igx-circular-bar--indeterminate', false);
```

Extract back the common class names used in the tests so that we can update them in a single place should the need arise in the future.

Reply:
I don't think this has been done.